controllers/kubeadmcontrolplane_controller.go

```diff
@@ -131,8 +128,10 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, log logr.
 		return ctrl.Result{}, err
 	}
 
-	if kcp.Spec.Replicas != nil && (*kcp.Spec.Replicas == kcp.Status.UpdatedReplicas) {
-		if cpuGetErr == nil && cpUpgrade.Status.Ready {
+	cpUpgrade := &anywherev1.ControlPlaneUpgrade{}
+	cpuGetErr := r.client.Get(ctx, GetNamespacedNameType(cpUpgradeName(kcp.Name), constants.EksaSystemNamespace), cpUpgrade)
```

Comment:
We aren't actually handling the case where cpuGetErr is nil and it is not the apierrors.IsNotFound error right?

Reply:
cpuGetErr == nil is being handled on line 133

Reply:
Yea that's what I meant, and ohh I didn't notice that. Should be fine though as even if we explicitly state, it would never really go that far.